src/main/java/com/linagora/james/mailets/GuessClassificationMailet.java

```diff
@@ -111,17 +132,47 @@ public void init() throws MessagingException {
         }
     }
 
+    private Optional<Integer> parseTimeout() throws MessagingException {
+        try {
+            Optional<Integer> result = Optional.ofNullable(getInitParameter(TIMEOUT_IN_MS))
+                .map(Integer::valueOf);
+            if (result.filter(value -> value < 1).isPresent()) {
+                throw new MessagingException("Non strictly positive timeout for " + TIMEOUT_IN_MS + ". Got " + getInitParameter(TIMEOUT_IN_MS));
+            }
+            return result;
+        } catch (NumberFormatException e) {
+            throw new MessagingException("Expecting " + TIMEOUT_IN_MS + " to be a strictly positive integer. Got " + getInitParameter(TIMEOUT_IN_MS));
+        }
+    }
+
     @Override
     public String getMailetInfo() {
         return "GuessClassificationMailet Mailet";
     }
 
     @Override
     public void service(Mail mail) throws MessagingException {
-        getClassificationGuess(mail)
+        Future<Optional<String>> predictionFuture = executorService.submit(() -> getClassificationGuess(mail));
+        awaitTimeout(predictionFuture)
             .ifPresent(classificationGuess -> addHeader(mail, classificationGuess));
     }
 
+    private Optional<String> awaitTimeout(Future<Optional<String>> objectFuture) {
```

Comment:
not sure we should do that right now, but in looking at alternative, I found that today, people use https://github.com/AsyncHttpClient/async-http-client for async http.